packages/language-server/src/server.ts

```diff
@@ -164,6 +175,34 @@ documents.onDidChangeContent(async (changeEvent) => {
   }
 });
 
+connection.onCodeAction((params) => {
+  const docUri = params.textDocument.uri;
+  const textDocument = documents.get(docUri);
+  if (textDocument === undefined) {
+    return undefined;
+  }
+
+  const diagnostics = params.context.diagnostics;
+  const codeActions = compact(
+    map(diagnostics, (_) => diagnosticToCodeActionFix(textDocument, _))
+  );
+
+  return codeActions;
+});
+
+connection.onExecuteCommand(async (params) => {
+  if (params.arguments === undefined) {
+    return;
+  }
+
+  const textDocument = documents.get(params.arguments[0]);
```

Comment:
- document assumption that all commands have textDocument URI in the first argument
- document doing runtime checks here.